src/sp/it/pl/service/hotkey/Hotkeys.kt

```diff
@@ -14,25 +14,21 @@ import java.util.concurrent.AbstractExecutorService
 import java.util.concurrent.ConcurrentHashMap
 import java.util.concurrent.TimeUnit
 
-/** Global hotkey manager, implemented on top of https://github.com/kwhat/jnativehook library. */
-class Hotkeys {
-    private val executor: (Runnable) -> Unit
+/** Global hotkey manager, implemented on top of JNativeHook library. */
+class Hotkeys(private val executor: (Runnable) -> Unit) {
```

Comment:
Please do not get into habit of removing constructors for gaining few lines of code. It causes field declaration fragmentation, which is just horrible horrible idea. I advise only using this when the primary constructors takes all public fields and pretty much no other case.

Reply:
IDEA suggested this, is it really so horrible?

Reply:
Not all suggestions are good all the time. Thats why it is possible to turn them off. For example if public method is only used in private context, IDA will complain that it can be private. Little can IDEA know about developer's intention to have that method as part of public API of that class so future developer can leverage it.

Now, I was overreacting with my words, but this style does lead to a hard to read mess.
What makes that one property special that it is defined in the primary constructor?
Why would a fact that a property value is passed by user affect where is the property declared??
Nobody would start defining fields at the bottom of the file right? Why would you define a field in a 'class' signature' then? It makes sense for value classes (a great lot), not object with state and behavior.

I would prefer this to be reverted and recommend this approach only for
- data classes
- classes with primary constructor defining all public fields, but preferably all fields.

An exception could be cases where overriding takes place, Id deal with that on case by case basis.

What do you think?

Reply:
This stayed, but I am now fairly convinced this is wrong approach, putting private fields in constructor, it is basically exposing them to the consumer. Constructor API and fields are separate thing!